### PR TITLE
Bloomfilter does not work with objects with hash == Int.MinValue 

### PR DESCRIPTION
**Pull request by wakoan**
Bloom Filter does not work for the next test

```scala
val bloomFilter = BloomFilter.optimallySized[Int](100, 0.03)
val value = Int.MinValue
bloomFilter += value
assert(bloomFilter.contains(value))
```

### CODE REVIEW

File: math/src/main/scala/breeze/util/BloomFilter.scala [hunk 1]
```diff
@@ -32,15 +32,21 @@ class BloomFilter[@specialized(Int, Long) T](val numBuckets: Int, val numHashFun
...
-      val hash = computeHash(h, hi)
-      if(!bits.get(hash % numBuckets)) return false
-      h+= 1
+  def activeBuckets(key: T) = {
+    val hash1 = key.##
+    val hash2 = math.abs(MurmurHash3.mixLast(0, hash1))
+
+    for {
+      i <- 0 to numHashFunctions
+      h = hash1 + i * hash2
```
Comment (dlwh) [inline math/src/main/scala/breeze/util/BloomFilter.scala]: in general I prefer assignments that are the end of the for comprehension to just be inside the body. creates fewer closures

**Comment by dlwh** [comment]
thanks a lot! I had one little nitpick.

**Comment by wakoan** [comment]
Fixed

**Comment by dlwh** [comment]
thanks!

**Pull request merged**